File: src/main/java/org/jukeboxmc/form/Modal.java
package org.jukeboxmc.form;

import com.nimbusds.jose.shaded.json.JSONObject;

/**
 * @author GoMint
 * @version 1.0
 */
public class Modal extends Form<Boolean> {

    private final String question;
    private String trueButtonText;
    private String falseButtonText;

    public Modal( String title, String question ) {
        super( title );
        this.question = question;
    }

    public void setTrueButtonText( String text ) {
        this.trueButtonText = text;
        this.dirty = true;
    }

    public void setFalseButtonText( String text ) {
        this.falseButtonText = text;
        this.dirty = true;
    }

    public String getFormType() {
        return "modal";
    }

    @Override
    public JSONObject toJSON() {
        // Fast out when cached
        if ( this.cache != null && !this.dirty ) {
            return this.cache;
        }

        // Create new JSON view of this form
        JSONObject jsonObject = super.toJSON();
        jsonObject.put( "content", this.question );
        jsonObject.put( "button1", this.trueButtonText );
        jsonObject.put( "button2", this.falseButtonText );

        // Cache and return
        this.cache = jsonObject;
        this.dirty = false;
        return this.cache;
    }

    @Override
    public Boolean parseResponse( String json ) {
        return json.trim().equals( "true" );
    }

}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/form/Modal.kt
package org.jukeboxmc.api.form

import com.google.gson.JsonObject

/**
 * @author Kaooot
 * @version 1.0
 */
class Modal(title: String, private val question: String) : Form<Boolean>(title) {

    private var trueButtonText: String? = null
    private var falseButtonText: String? = null

    override fun getFormType(): String = "modal"

    override fun parseResponse(json: String): Boolean = json.trim() == "true"

    override fun toJson(): JsonObject {
        if (this.cache != null && !this.dirty) {
            return this.cache!!
        }
        val jsonObject = super.toJson()
        jsonObject.addProperty("content", this.question)
        jsonObject.addProperty("button1", this.trueButtonText)
        jsonObject.addProperty("button2", this.falseButtonText)
        this.cache = jsonObject
        this.dirty = false
        return this.cache!!
    }

    fun setTrueButtonText(text: String) {
        this.trueButtonText = text
        this.setDirty()
    }

    fun setFalseButtonText(text: String) {
        this.falseButtonText = text
        this.setDirty()
    }
}